src/features/Applet/AppletSettings/ReportConfigSetting/ReportConfigSetting.tsx

```diff
@@ -22,39 +24,52 @@ import { StyledButton, StyledSvg, StyledContainer, StyledForm } from './ReportCo
 import { FormValues } from './ReportConfigSetting.types';
 
 export const ReportConfigSetting = () => {
+  const { id } = useParams();
+  const applet = folders.useApplet(id as string);
   const { t } = useTranslation();
   const isServerConfigured = false; // TODO: add server configured functionality when the back-end is ready
   const [isSettingsOpen, setSettingsOpen] = useState(false);
-  const {
-    handleSubmit,
-    control,
-    setValue,
-    formState: { isValid },
-    watch,
-  } = useForm<FormValues>({
+  const { handleSubmit, control, setValue, watch, trigger } = useForm<FormValues>({
     resolver: yupResolver(reportConfigSchema()),
     defaultValues,
-    mode: 'onChange',
+    mode: 'onSubmit',
   });
-  const emails = watch('emails');
+  const emailRecipients = watch('emailRecipients');
+  const respondentId = watch('respondentId');
+  const caseId = watch('caseId');
+
+  const handleAddEmail = async (value: string) => {
+    const isValid = await trigger(['email']);
+    if (!isValid) return;
 
-  const handleAddEmail = (value: string) => {
     if (value.length) {
       setValue('email', '');
-      if (!emails.some((item) => item === value)) {
-        setValue('emails', [...emails, value]);
+      if (!emailRecipients.some((item) => item === value)) {
+        setValue('emailRecipients', [...emailRecipients, value]);
       }
     }
   };
 
   const handleRemoveEmail = (index: number) => {
-    setValue('emails', emails.filter((_, i) => i !== index) as string[]);
+    setValue('emailRecipients', emailRecipients.filter((_, i) => i !== index) as string[]);
   };
 
   const onSubmit = (values: FormValues) => {
     console.log(values);
   };
 
+  useEffect(() => {
+    let subject = 'REPORT';
+    if (respondentId) {
+      subject += ' by user123';
+    }
+    if (caseId) {
+      subject += ' about case123';
+    }
+    subject += `: ${applet.name || 'Example applet'} /activity123 or activityflow123`; // TODO will be fixed for activities
```

Comment:
user123, case123, activity123 or activityflow123 - hardcoded values (according to old version), they will be replaced to real data on the sever